### PR TITLE
Give Interval a Curve to apply between start and end

### LINKED ISSUE #1168: Combine AnimationTiming.interval/reverseInterval and AnimationTiming.curve/reverseCurve
Drop AnimationTiming.interval/reverseInterval, and instead use AnimationTiming.curve/reverseCurve with a CompoundCurve() that takes two curves and chains them.

We could also merge curve/reverseCurve into a single one if we passed the direction in transform().

### PR DESCRIPTION
**Pull request by abarth**
This patch simplifies AnimationTiming and all the AnimatedValue base classes.
Also, make PopupMenu a stateless component because it has no state.

Fixes #1168

### CODE REVIEW

File: sky/packages/sky/lib/src/animation/animated_value.dart [hunk 1]
```diff
@@ -25,70 +25,43 @@ abstract class Animatable {
...
-  /// The interval during which this timing is active in the forward direction
-  Interval interval;
-
-  /// The interval during which this timing is active in the reverse direction
-  ///
-  /// If this field is null, the timing defaults to using [interval] in both directions.
-  Interval reverseInterval;
-
-  /// The curve that this timing applies to the animation clock in the forward direction
+  /// The curve to use in the forward direction
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/animated_value.dart]: (nit: add periods at the end of dart docs when you come across those that don't have them. don't worry about doing it in this PR necessarily)

File: sky/packages/sky/lib/src/widgets/animated_container.dart [hunk 2]
```diff
@@ -9,29 +9,29 @@ import 'package:sky/src/widgets/framework.dart';
 import 'package:vector_math/vector_math_64.dart';
 
 class AnimatedBoxConstraintsValue extends AnimatedValue<BoxConstraints> {
-  AnimatedBoxConstraintsValue(BoxConstraints begin, { BoxConstraints end, Curve curve: linear })
-    : super(begin, end: end, curve: curve);
+  AnimatedBoxConstraintsValue(BoxConstraints begin, { BoxConstraints end, Curve curve, Curve reverseCurve })
```
Comment (Hixie) [inline sky/packages/sky/lib/src/widgets/animated_container.dart]: why is changing the default here ok?

Comment (abarth) [inline sky/packages/sky/lib/src/widgets/animated_container.dart]: Passing `null` for `curve` does the same thing as linear.  The default here didn't do anything.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**